test/errors.js

```diff
@@ -0,0 +1,158 @@
+var Pipeline = require( '..' )
+var assert = require( 'assert' )
+var Stream = require( 'stream' )
+
+describe( 'Pipeline#onError', function() {
+
+  it( 'will emit an "error" from the beginning of the pipeline', function( done ) {
+
+    var stream1 = new Stream.PassThrough()
+    var stream2 = new Stream.Transform({
+      transform: function( chunk, _, next ) {
+        next( null, chunk.toString().toLowerCase() )
+      }
+    })
+
+    var stream3 = new Stream.Transform({
+      transform: function( chunk, _, next ) {
+        next( null, chunk.toString().toUpperCase() )
+      }
+    })
+
+    var pipeline = new Pipeline([ stream1, stream2, stream3 ])
+    var hadError = false
+
+    pipeline.on( 'error', function( error ) {
+      assert.ok( hadError === false, 'only one error emitted' )
+      assert.equal( error.stream, stream1, 'stream reference' )
+      hadError = true
+    })
+
+    pipeline.resume().once( 'finish', function() {
+      done()
+    })
+
+    pipeline.write( 'something' )
+    pipeline.end( 'else' )
+
+    setImmediate( function() {
+      stream1.emit( 'error', new Error( 'Catch me' ) )
+    })
+
+  })
+
+  it( 'will emit every error from all streams in the pipeline', function( done ) {
```

Comment:
Well, it will actually re-emit every error, not just one. Stream semantics are a little undefined around errors; some will only emit one, or won't emit errors after a certain point, etc. So I figured it's probably best to just emit them as they come.

Yup, quoted the "error" for consistency now.